modules/standard/Random.chpl

```diff
@@ -167,24 +106,41 @@ record SeedGenerators {
   :arg seed: The seed to use for the PRNG.  Defaults to :proc:`SeedGenerator.currentTime <SeedGenerators.currentTime>`.
   :type seed: int(64)
 */
-
-proc fillRandom(arr: [], seed: int(64) = SeedGenerator.currentTime)
-  where (arr.eltType == real || arr.eltType == imag || arr.eltType == complex) {
-  var randNums = new RandomStream(seed, eltType=arr.eltType, parSafe=false);
+proc fillRandom(arr: [], seed: int(64) = SeedGenerator.currentTime, param
+    algorithm=defaultRNG)
+  where isSupportedNumericType(arr.eltType) {
+  var randNums = makeRandomStream(seed, eltType=arr.eltType, parSafe=false, algorithm=algorithm);
   randNums.fillRandom(arr);
   delete randNums;
 }
 
 pragma "no doc"
-proc fillRandom(arr: [], seed: int(64) = SeedGenerator.currentTime) {
-  compilerError("Random.fillRandom is only defined for real(64), imag(64), and complex(128) arrays");
+proc fillRandom(arr: [], seed: int(64) = SeedGenerator.currentTime, param
+    algorithm=defaultRNG) {
+  compilerError("Random.fillRandom is only defined for numeric arrays");
+}
+
+// shuffle might also be a bad name
+proc shuffle(arr: [], seed: int(64) = SeedGenerator.currentTime, param algorithm=RNG.PCG) {
+  var randNums = makeRandomStream(seed, eltType=arr.eltType, parSafe=false, algorithm=algorithm);
+  randNums.shuffle(arr);
+  delete randNums;
+}
+
+/// permutation is a poor name
+proc permutation(arr: [], seed: int(64) = SeedGenerator.currentTime, param algorithm=RNG.PCG) {
```

Comment:
Are permutation and shuffle part of the official PCG interface?  Otherwise I'm not sure why they were added as a part of this PR vs. independent from it.  Also, the algorithm=RNG.NPB version doesn't exist.